pkg/proto/v0/settings.pb.micro_test.go

```diff
@@ -690,4 +811,194 @@ func TestCreateRoleAndAssign(t *testing.T) {
+				Resource: &proto.Resource{
+					Type: proto.Resource_TYPE_SYSTEM,
+				},
+			}},
+			expectedBundles: []expectedBundle{
+				{displayName: "test role - update", name: "TEST_ROLE"},
+			},
+		},
+		{name: "two added bundles",
+			bundles: []*proto.Bundle{{
+				Type:        proto.Bundle_TYPE_ROLE,
+				DisplayName: "test role - update",
+				Name:        "TEST_ROLE",
+				Extension:   "ocis-settings",
+				Settings: []*proto.Setting{
+					{
+						Name: "settingName",
+						Resource: &proto.Resource{
+							Id:   settingsStub[0].Id,
+							Type: proto.Resource_TYPE_SETTING,
+						},
+						Value: &proto.Setting_PermissionValue{
+							PermissionValue: &proto.Permission{
+								Operation:  proto.Permission_OPERATION_UPDATE,
+								Constraint: proto.Permission_CONSTRAINT_OWN,
+							},
+						},
+					},
+				},
+				Resource: &proto.Resource{
+					Type: proto.Resource_TYPE_SYSTEM,
+				},
+			},
+				{
+					Type:        proto.Bundle_TYPE_ROLE,
+					DisplayName: "an other role",
+					Name:        "AnOtherROLE",
+					Extension:   "ocis-settings",
+					Settings: []*proto.Setting{
+						{
+							Name: "settingName",
+							Resource: &proto.Resource{
+								Id:   settingsStub[0].Id,
+								Type: proto.Resource_TYPE_SETTING,
+							},
+							Value: &proto.Setting_PermissionValue{
+								PermissionValue: &proto.Permission{
+									Operation:  proto.Permission_OPERATION_UPDATE,
+									Constraint: proto.Permission_CONSTRAINT_OWN,
+								},
+							},
+						},
+					},
+					Resource: &proto.Resource{
+						Type: proto.Resource_TYPE_SYSTEM,
+					},
+				}},
+			expectedBundles: []expectedBundle{
+				{displayName: "test role - update", name: "TEST_ROLE"},
+				{displayName: "an other role", name: "AnOtherROLE"},
+			},
+		},
+	}
+
+	rebuidDataStore()
+	client := service.Client()
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			roleService := proto.NewRoleService("com.owncloud.api.settings", client)
+
+			bundleService := proto.NewBundleService("com.owncloud.api.settings", client)
+			for _, bundle := range tt.bundles {
+				_, err := bundleService.SaveBundle(context.Background(), &proto.SaveBundleRequest{
+					Bundle: bundle,
+				})
+				assert.NoError(t, err)
+			}
+			rolesRes, err := roleService.ListRoles(context.Background(), &proto.ListBundlesRequest{AccountUuid: "."})
+			assert.NoError(t, err)
+
+			for _, bundle := range rolesRes.Bundles {
+				assert.Contains(t, tt.expectedBundles, expectedBundle{
+					displayName: bundle.DisplayName,
+					name:        bundle.Name,
+				})
+			}
+			assert.Equal(t, len(tt.expectedBundles), len(rolesRes.Bundles))
+
+			os.RemoveAll(dataStore)
+		})
+	}
+}
+
+func TestListRolesVariousAccountUuid(t *testing.T) {
+	tests := []struct {
+		name        string
+		accountUUID string
+		error       interface{}
+	}{
+		{"space",
+			" ",
+			"{\"id\":\"ocis-settings\",\"code\":400,\"detail\":\"must be in a valid format\",\"status\":\"Bad Request\"}",
```

Comment:
We're using the `go-micro` errors in our handlers now. You can define expected errors in tests like this:
`merrors.New("ocis-settings", "must be in a valid format", 400),`
@individual-it